lib/jobs/boot-profile.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = bootProfileJobFactory;
+di.annotate(bootProfileJobFactory, new di.Provide('Job.BootProfile'));
+di.annotate(bootProfileJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'Logger',
+        'Assert',
+        'Util'
+    )
+);
+function bootProfileJobFactory(BaseJob, Logger, assert, util) {
+    var logger = Logger.initialize(bootProfileJobFactory);
+
+    /**
+     * NOTE: Due to the way our boilerplate.ipxe script works, this job
+     * will not work if run as the final task within a workflow (boilerplate.ipxe
+     * causes ipxe to make additional DHCP requests prior to actually
+     * running the ipxe script we've served down, and those requests will get
+     * ignored if there is no currently active workflow for the node.
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function BootProfileJob(options, context, taskId) {
+        BootProfileJob.super_.call(this, logger, options, context, taskId);
+        this.nodeId = this.context.target;
+        assert.string(this.nodeId, 'context.target (nodeId)');
+        this.profile = this.options.profile;
+        assert.string(this.profile, 'options.profile');
+    }
+    util.inherits(BootProfileJob, BaseJob);
+
+    BootProfileJob.prototype._run = function () {
+        var self = this;
+
+        self._subscribeRequestProfile(function () {
+            return self.profile;
+        });
+
+        self._subscribeRequestProperties(function () {
+            self._done();
```

Comment:
@yyscamper the behavior of the HTTP server when serving profiles is to request a profile name first, then request properties that can be used for the render context for the profile (see https://github.com/RackHD/on-http/blob/master/lib/services/profiles-api-service.js#L257-L266). If we listen for requestProfile only, then http will fail on the requestProperties step. Additionally, this leaves the possibility open for a user to add a profile that requires some rendering context from the task.

I understand your concern about a restriction on the behavior of the upper layer, but it's pretty much the standard way we do this throughout the http profiles code and the job code. Maybe a better way to structure this would be refactoring requestProfile to return both a profile name and a render context all in one call, which would be less complex and impose fewer restrictions on the upper layer, but I also think this is just a pattern of behavior/technical debt that's pretty ingrained into the system that might not be worth the effort of replacing given how long it has been around.